src/main/scala/org/zalando/kanadi/api/Subscriptions.scala

```diff
@@ -756,10 +762,11 @@ case class Subscriptions(baseUri: URI, oAuth2TokenProvider: Option[OAuth2TokenPr
                       toHeader(oAuth2Token) +: streamHeaders
                     }
                 }
-      entity   <- Marshal(subscriptionCursor).to[RequestEntity]
-      request  = HttpRequest(HttpMethods.POST, uri, headers, entity)
-      _        = logger.debug(request.toString)
-      response <- http.singleRequest(request).map(decodeCompressed)
+      entity  <- Marshal(subscriptionCursor).to[RequestEntity]
+      request = HttpRequest(HttpMethods.POST, uri, headers, entity)
+      _       = logger.debug(request.toString)
+      response <- if (eventBatch) http.singleRequest(request).map(decodeCompressed)
```

Comment:
If you don't read from the stream then you need to discard it, use `response.entity.dataBytes.runWith(Sink.ignore)`

Reply:
This is supposed to be done by the next section as the “keep-alive” case is just mapped to `204 No content` directly. Or do I miss something else here?